routes/awsapi.js

```diff
@@ -0,0 +1,132 @@
+
+if(ACCESS_KEY_ID == null || SECRET_ACCESS_KEY == null){
+  console.log("ERROR: s3 envirnoment variables not set!");
+}
+
+aws.config.update({
+  accessKeyId: ACCESS_KEY_ID,
+  secretAccessKey: SECRET_ACCESS_KEY
+});
+
+
+module.exports = function (io, db) {
+  router.route('/s3-sign')
+    .get(GETSignRequest)
+    .post(POSTS3Sign);
+
+  router.route('/MediaStatusSNS')
+    .post(POSTSNSNotify);
+
+  /*
+   * Respond to GET requests to /sign-s3.
+   * Upon request, return JSON containing the temporarily-signed S3 request and
+   * the anticipated URL of the image.
+   */
+  function GETSignRequest(req, res) {
+    const s3 = new aws.S3({
+        signatureVersion: 'v4',
+        // Endpoint can be switched to https://media.futureboard.olin.build/' after we get an ssl cert for media.FUTUREboard.olin.build
+        endpoint: new aws.Endpoint('https://s3.amazonaws.com')
+      });
+    const inputFileName = req.query['file-name'].split('.');
+    const fileType = req.query['file-type'];
+    const mediaType = fileType.split('/')[0]; // Get image or video
+    const inputFileExtension = inputFileName[inputFileName.length - 1];
+    let uploadFileName = '';
+    let outputFileName = '';
+    const new_uuid = uuidv1();
+
+    if(mediaType === 'image'){
+      uploadFileName = 'img-' + new_uuid + '.' + inputFileExtension;
+      outputFileName = 'img-' + new_uuid + '.jpg';
+    } else if (mediaType === 'video') {
+      uploadFileName = 'vid-' + new_uuid + '.' + inputFileExtension;
+      outputFileName = 'vid-' + new_uuid + '.mp4';
+    } else {
+      res.status(415).send('unsuported media type');
+      return;
+    }
+    const s3Params = {
+      Bucket: S3_BUCKET,
+      Key: uploadFileName,
+      Expires: 60,
+      ContentType: fileType,
+      ACL: 'public-read'
+    };
+
+    s3.getSignedUrl('putObject', s3Params, function (err, data) {
+      if(err){
+        console.log(err);
+        return res.end();
+      }
+      const returnData = {
+        signedRequest: data,
+        // The location of the future media, to be used for previering and submitting a bob
+        url: 'http://media.futureboard.olin.build/' + outputFileName
+      };
+      res.write(JSON.stringify(returnData));
+      res.end();
+    });
+  }
+
+  function POSTS3Sign(req, res) {
+    res.send("success");
+  }
+
+  function POSTSNSNotify (req, res) {
+    // Recieve all data
+    var chunks = [];
+    req.on('data', function (chunk) {
+        chunks.push(chunk);
+    });
+    // After all data is received
+    req.on('end', function () {
+        try {
+          var message = JSON.parse(chunks.join(''));
+          // If it is a subscribtion confirmation, get the page
+          if(req.headers['x-amz-sns-message-type'] === 'SubscriptionConfirmation'){
+            request(message.SubscribeURL, function (err, res, body) {
+              if(err){ console.log(err); }
+            });
+          // Else set the bob media status to true
+          } else {
```

Comment:
Might be good to specify `req.headers['x-amz-sns-message-type']` here too so that we don't accidentally try to use some other type of message as a confirmation